source/app.d

```diff
@@ -165,7 +165,26 @@ shared static this()
 	auto urlRouter = new URLRouter;
 	auto fsettings = new HTTPFileServerSettings;
 	fsettings.serverPathPrefix = "/static";
+
+	void cors(HTTPServerRequest req, HTTPServerResponse res)
+	{
+		import std.algorithm: among, equal, until;
+		if (req.host.until(":").among!equal("tour.dlang.org", "tour.dlang.io", "dlang.org", "dtest.dlang.io", "127.0.0.1", "localhost"))
+		{
+			res.headers["Access-Control-Allow-Origin"] = "*";
+		}
+		// parse json sent via text/plain to avoid an additional preflight OPTIONS request
+		// https://developer.mozilla.org/en-US/docs/Web/HTTP/Access_control_CORS#Preflighted_requests
+		if (req.contentType == "text/plain")
+		{
+			req.contentType = "application/json; charset=UTF-8";
+			auto bodyStr = () @trusted { return cast(string)req.bodyReader.readAll(); } ();
+			if (!bodyStr.empty) req.json = parseJson(bodyStr);
```

Comment:
Basically what https://github.com/rejectedsoftware/vibe.d/blob/67ac2cf81f73854e00304c24aca5c889a627d6f4/http/vibe/http/server.d#L2016 does.

After the 0.8.0 release, I hope that my PR for lazy initialization hopefully gets accepted (https://github.com/rejectedsoftware/vibe.d/pull/1677), s.t. the manual parsing won't needed anymore.